### PR TITLE
doc: guidelines for authoring tests

### PR DESCRIPTION
**Pull request by ghuntley**
## PR Type

What kind of change does this PR introduce?

- Documentation content changes

## Other information

Opening up as a draft pull-request to increase discoverability.

### CODE REVIEW

File: doc/articles/contributing/guidelines/testing.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# Guidelines for authoring tests
```
Comment (ghuntley) [inline doc/articles/contributing/guidelines/testing.md]: 🎗 need to add to the TOC

File: doc/articles/contributing/guidelines/testing.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+## integration testing
+
+### Concepts
+
+- `Constants.cs`
+- `Run(xxxx)` specifics the full class path to the control
+- `SamplesApp*`
+- https://github.com/unoplatform/Uno.UITest
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/testing.md]:
```suggestion
- https://github.com/unoplatform/uno.uitest
```

File: doc/articles/contributing/guidelines/testing.md (hunk 1 above)
Comment (jeromelaban) [reply doc/articles/contributing/guidelines/testing.md]: Merge or pull content from https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md

File: doc/articles/contributing/guidelines/testing.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+### Dev Loop
+
+1. Start emulators and deploy application (iOS and Android)
+1. Start without debugging to ensure the webserver is running (webasm)
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/testing.md]:
```suggestion
1. Start without debugging to ensure the webserver is running (Wasm)
```

It's either _WebAssembly_ or _Wasm_ (this specific casing and spaces): See https://webassembly.org: 
image

File: doc/articles/contributing/guidelines/testing.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+1. Adjust https://github.com/unoplatform/uno/blob/dev/ghuntley/doc/testing-guidelines/src/SamplesApp/SamplesApp.UITests/Constants.cs to specify which platform to test
+1. Run the test via test explorer (or resharper)
+
+### See Also
+
+- https://github.com/unoplatform/uno/blob/dev/ghuntley/doc/testing-guidelines/src/SamplesApp/SamplesApp.UITests/Constants.cs
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/testing.md]: Link in master ?

File: doc/articles/contributing/guidelines/testing.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+			var addButton = _app.Marked("AddButton");
+			var containsButton = _app.Marked("ContainsButton");
+			var removeButton = _app.Marked("RemoveButton");
+			var output = _app.Marked("Output");
+
+			// Specify what user interface element to wait on before starting test execution
+			_app.WaitForElement(clearButton);
+
+			// Take an initial screenshot
+			_app.Screenshot(nameof(ClearAddContainsRemove) + " - Initial State");
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/testing.md]:
```suggestion
			TakeScreenshot("Initial State");
```

See https://github.com/unoplatform/uno/blob/a6db376e46afd9fb30dee317f08a2e50dc5310e4/src/SamplesApp/SamplesApp.UITests/SampleControlUITestBase.cs#L89 

Which automatically includes the current test information

File: doc/articles/contributing/guidelines/testing.md [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+
+			{
+				_app.Tap(containsButton);
+				_app.Screenshot(nameof(ClearAddContainsRemove) + " - Contains Button");
+				_app.WaitForDependencyPropertyValue(output, "Text", "True");
+			}
+
+			{
+				_app.Tap(removeButton);
+				_app.Screenshot(nameof(ClearAddContainsRemove) + " - Remove Button");
```
Comment (jeromelaban) [inline doc/articles/contributing/guidelines/testing.md]: Screenshot should assert the result after the `WaitForDependencyPropertyValue` when the state is stable, as the Tap may not react immediately. Same for the others above.

File: doc/articles/contributing/guidelines/testing.md (hunk 1 above)
Comment (ghuntley) [reply doc/articles/contributing/guidelines/testing.md]: Let's:
- Burn the content of this file.
- Rename working-with-the-samples-apps.md to `doc/articles/contributing/guidelines/testing.md`
- Move all examples for code style into a separate document that contains all conventions -  https://github.com/unoplatform/uno/pull/1212/files

File: doc/articles/contributing/guidelines/testing.md (hunk 6 above)
Comment (ghuntley) [reply doc/articles/contributing/guidelines/testing.md]: Resolved over at https://github.com/unoplatform/uno/pull/1212/files#diff-336e4bcde45e4393a322cd34f381311d

**Comment by davidjohnoliver** [comment]
The rename makes sense, but this'll break links to the generated html. You might want to add back the old filename with a

**Pull request closed without merging**